File: backend/app/schemas/transaction.py
from pydantic import BaseModel
from datetime import datetime
from app.models.transaction import TipeTransaksi
from typing import Optional

class TransactionCreate(BaseModel):
    judul: str
    nominal: float
    tipe: TipeTransaksi
    kategori: str
    catatan: Optional[str] = None

class TransactionResponse(BaseModel):
    id: int
    user_id: int
    judul: str
    nominal: float
    tipe: TipeTransaksi
    kategori: str
    catatan: Optional[str]
    tanggal: datetime

    class Config:
        from_attributes = True

File: apps/mobile/lib/models/transaction.dart
class Transaction {
  final int id;
  final int userId;
  final String judul;
  final double nominal;
  final String tipe;
  final String kategori;
  final String? catatan;
  final DateTime tanggal;

  Transaction({
    required this.id,
    required this.userId,
    required this.judul,
    required this.nominal,
    required this.tipe,
    required this.kategori,
    this.catatan,
    required this.tanggal,
  });

  factory Transaction.fromJson(Map<String, dynamic> json) {
    return Transaction(
      id: json['id'],
      userId: json['user_id'],
      judul: json['judul'],
      nominal: json['nominal'].toDouble(),
      tipe: json['tipe'],
      kategori: json['kategori'],
      catatan: json['catatan'],
      tanggal: DateTime.parse(json['tanggal']),
    );
  }
}